src/app/components/jsx-helpers/jsx-helpers.jsx

```diff
@@ -1,4 +1,34 @@
-import React from 'react';
+import React, {useState, useLayoutEffect} from 'react';
+
+function getValuesFromWindow() {
+    const {innerHeight, innerWidth, scrollY} = window;
+
+    return {innerHeight, innerWidth, scrollY};
+}
+
+export const WindowContext = React.createContext(getValuesFromWindow());
+
+export function WindowContextProvider({children}) {
+    const [value, setValue] = useState(getValuesFromWindow());
+
+    useLayoutEffect(() => {
+        const handleScroll = () => setValue(getValuesFromWindow());
+
```

Comment:
the removeListeners looks great, but I think you'll also want to wrap the `setValue` function with a `debounce` so it isn't called too often.